tests/queries/0_stateless/02242_make_date.sql

```diff
@@ -0,0 +1,69 @@
+select toTypeName(makeDate(1991, 8, 24));
+select toTypeName(makeDate(cast(1991 as Nullable(UInt64)), 8, 24));
+select toTypeName(makeDate(1991, cast(8 as Nullable(UInt64)), 24));
+select toTypeName(makeDate(1991, 8, cast(24 as Nullable(UInt64))));
+select toTypeName(makeDate(1991, cast(8 as Nullable(UInt64)), cast(24 as Nullable(UInt64))));
+
+select makeDate(1970, 01, 01);
+select makeDate(2020, 08, 24);
+select makeDate(1980, 10, 17);
+select makeDate(-1980, 10, 17);
+select makeDate(1980, -10, 17);
+select makeDate(1980, 10, -17);
+select makeDate(1980.0, '10', 30.0/2);
+select makeDate(-1980.0, '10', 30.0/2);
+select makeDate(cast(1980.1 as Decimal(20,5)), '10', 30.0/2);
+select makeDate(cast('-1980.1' as Decimal(20,5)), '10', 30.0/2);
+select makeDate(cast(1980.1 as Float32), '10', 30.0/2);
+select makeDate(cast(-1980.1 as Float32), '10', 30.0/2);
+--select makeDate(cast(1980 as Date), '10', 30.0/2);
+--select makeDate(cast(-1980 as Date), '10', 30.0/2);
+--select makeDate(cast(1980 as Date32), '10', 30.0/2);
+--select makeDate(cast(-1980 as Date32), '10', 30.0/2);
+--select makeDate(cast(1980 as DateTime), '10', 30.0/2);
+--select makeDate(cast(-1980 as DateTime), '10', 30.0/2);
+--select makeDate(cast(1980 as DateTime64), '10', 30.0/2);
+--select makeDate(cast(-1980 as DateTime64), '10', 30.0/2);
+
+select makeDate(0.0, 1, 2);
+select makeDate(1980, 15, 1);
+select makeDate(1980, 2, 29);
+select makeDate(1984, 2, 30);
+select makeDate(19800, 12, 3);
+select makeDate(2148,1,1);
+select makeDate(2149,1,1);
+select makeDate(2149,6,6);
+select makeDate(2149,6,7);
+select makeDate(2150,1,1);
+select makeDate(1969,1,1);
+select makeDate(1969,12,1);
+select makeDate(1969,12,31);
+select makeDate(1970,0,0);
+select makeDate(1970,0,1);
+select makeDate(1970,1,0);
+select makeDate(1990,0,1);
+select makeDate(1990,1,0);
+
+select makeDate('1980', '10', '20');
```

Comment:
So now even something like `makeDate('1980.05', '10', '20.3')` should work?

Reply:
I replaced my custom to conversion logic to castColumn() that is used in other similar cases.
Now string arguments will not be accepted, only numbers will be allowed.